apps/meteor/app/livechat/server/lib/Livechat.js

```diff
@@ -571,14 +574,14 @@ export const Livechat = {
 		const { _id } = user;
 
 		if (await removeUserFromRolesAsync(_id, ['livechat-agent'])) {
-			await Users.setOperator(_id, false);
-			await Users.removeLivechatData(_id);
-			await this.setUserStatusLivechat(_id, 'not-available');
-
 			await Promise.all([
+				Users.setOperator(_id, false),
+				Users.removeLivechatData(_id),
+				Users.unsetLivechatStatus(_id),
 				LivechatDepartmentAgents.removeByAgentId(_id),
 				LivechatVisitors.removeContactManagerByUsername(username),
 				Users.unsetExtension(_id),
+				Users.removeAllBusinessHoursFromUserByUserId(_id),
```

Comment:
there are 5 operations happening to the same user 🙈 can we improve this and do all updates in a single command?